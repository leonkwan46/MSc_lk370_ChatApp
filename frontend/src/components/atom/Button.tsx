import React, { type FC } from 'react'
import { StyleProp, TouchableOpacity, ViewStyle } from 'react-native'
import { getButtonStyle } from '@/src/helpers/styleHelper'

interface ButtonProps {
    id?: string
    children: React.ReactNode
    color?: string
    size?: string
    fill?: boolean
    extraStyles?: object
    onPress: () => void
}

const Button: FC<ButtonProps> = ({
    id='',
    children,
    color='',
    size='',
    fill,
    extraStyles,
    onPress
}) => {
    const { containerStyle } = getButtonStyle({ size, color, fill })
    const style: StyleProp<ViewStyle> = [containerStyle, extraStyles]

    return (
        <TouchableOpacity id={id} style={style} onPress={onPress}>
            {children}
        </TouchableOpacity>
    )
}

export default Button