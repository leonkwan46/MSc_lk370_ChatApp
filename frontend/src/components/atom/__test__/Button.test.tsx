import * as React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

it('renders correctly with default props', () => {
  const tree = renderer
    .create(<Button onPress={() => {}}>Snapshot test!</Button>)
    .toJSON()
  
  expect(tree).toMatchSnapshot()
})

it('renders correctly with color, size, and fill props', () => {
  const tree = renderer
    .create(
      <Button color="blue" size="large" fill onPress={() => {}}>
        Snapshot test with props!
      </Button>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('renders correctly with extraStyles', () => {
  const extraStyles = { backgroundColor: 'green', padding: 10 }
  const tree = renderer
    .create(
      <Button onPress={() => {}} extraStyles={extraStyles}>
        Styled Button
      </Button>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
