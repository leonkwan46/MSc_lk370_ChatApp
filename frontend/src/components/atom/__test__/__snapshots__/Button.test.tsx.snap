// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with color, size, and fill props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderColor": "",
      "borderRadius": 5,
      "borderWidth": 0,
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 18,
      "paddingVertical": 10,
      "width": "50%",
    }
  }
>
  Snapshot test with props!
</View>
`;

exports[`renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderColor": "",
      "borderRadius": 5,
      "borderWidth": 0,
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 18,
      "paddingVertical": 10,
      "width": "50%",
    }
  }
>
  Snapshot test!
</View>
`;

exports[`renders correctly with extraStyles 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderColor": "",
      "borderRadius": 5,
      "borderWidth": 0,
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
      "paddingHorizontal": 18,
      "paddingVertical": 10,
      "width": "50%",
    }
  }
>
  Styled Button
</View>
`;
